modules/blockexplorer/blockexplorer.go

```diff
@@ -0,0 +1,55 @@
+package blockexplorer
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/sync"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// The blockexplorer module provides a glimpse into what the blockchain
+// currently looks like.
+
+// Basic structure to store the blockchain. Metadata may also be
+// stored here in the future
+type ExplorerState struct {
+	// CurBlock is the current highest block on the blockchain,
+	// kept update via a subscription to consensus
+	currentBlock types.Block
+
+	// Used for caching the current blockchain height
+	blockchainHeight types.BlockHeight
+
+	mu *sync.RWMutex
+}
+
+// New creates the internal data structures, and subscribes to
+// consensus for changes to the blockchain
+func New(cs modules.ConsensusSet) (bc *ExplorerState, err error) {
+
+	// Check that input modules are non-nil
+	if cs == nil {
+		err = errors.New("Blockchain explorer cannot use a nil ConsensusSet")
+		return
+	}
+
+	// Initilize the module state
+	bc = &ExplorerState{
+		currentBlock:     cs.CurrentBlock(),
+		blockchainHeight: 0,
```

Comment:
I don't think either implementation accounts for this, but what would happen if consensus finds blocks and publishes them to subscribers before this module is initialized and has a chance to subscribe to consensus?